### DISCUSSION TITLE
想请教下Adguardhome+openclash的组合如何解析ipv6？

### DISCUSSION THREAD

**Discussion by zzz6839** [vernesong/OpenClash#2164] [Q&A] [answered]
AdGuardHome
重定向模式：DNSMASQ的上游服务器
上游DNS服务器：127.0.0.1:7874
Openclash
IPV6流量代理✔
允许IPv6类型DNS解析✔
设置自定义上游DNS服务器：有ipv6的fallback和NS在53口
Openwrt/网络
DHCP/DNS：允许解析 IPv6 DNS 记录

IPV6WAN状态
类型: dhcpv6-pd
分发前缀: 240e:390:80f:2850::/60
地址: 240e:390:8c1:b031:6473:98cd:ecbd:8d6/64
网关: fe80::225:9eff:fe81:fa56
已连接: 1h 12m 32s

Lan
路由通告服务
服务器模式
DHCPv6 服务
服务器模式
NDP 代理
已禁用 
DHCPv6 模式
无状态的 + 有状态的
总是通告默认路由

终端设备

IPv6 Address: 240e:390:80f:2850::a9f
Lease Obtained: Saturday, March 19, 2022 6:59:47 PM
Lease Expires: Sunday, March 20, 2022 6:59:47 AM, 240e:390:80f:2850:8567:b821:4f24:77c0
Temporary IPv6 Address: 240e:390:80f:2850:bccf:cc5f:31ba:4f63
Link-local IPv6 Address: fe80::8567:b821:4f24:77c0%22
IPv6 Default Gateway: fe80::2e2:69ff:fe24:7a04%22
IPv6 DNS Server: 240e:390:80f:2850::1

以上
通过了ipv6测试和测速网站测试。

但是在openclash里的日志中从来没见过ipv6的域名解析过，有心血来潮把::1:7874设为adguardhome上游服务器，观察到局域网下设备的dhcpv6分配的ip，但是没有解析ipv6的域名，只有直连过ipv6的ip，所以想请教下该怎么改，谢谢大神

**Comment by vernesong** [comment, marked as answer]
下面设备v6的网关和DNS要改过去，

**Reply by zzz6839** [reply]
感谢回复，DNS现在是指向路由器的所以没有问题，就是不知道IPV6的网关该怎么改了，openwrtX86

**Reply by vernesong** [reply]
路由器lan的v6地址啊

**Reply by tolshao** [reply]
同不懂
